### PR TITLE
Getting logits to the client code

### PR DESCRIPTION
**Pull request by adeepH**
Follow up to #2075, and #2340, to return the model outputs to the user.

### CODE REVIEW

**Comment by ftyers** [comment]
Hi @reuben can you check this if you have a moment? Thanks!

**Comment by ftyers** [comment]
Thanks @reuben !
